zzk/service/hoststate.go

```diff
@@ -157,25 +157,16 @@ func (l *HostStateListener) Spawn(shutdown <-chan interface{}, stateID string) {
 
 		// load the service state node
 		sspth := path.Join("/services", serviceID, stateID)
-		ssdat := &ServiceState{}
-		if err := l.conn.Get(sspth, ssdat); err == client.ErrNoNode {
-
+		exists, err := l.conn.Exists(sspth)
```

Comment:
You do need to call Get on ServiceState on first start so that it knows if it is attaching to an existing container.